smoke/internal/util.go

```diff
@@ -0,0 +1,349 @@
+			require.Contains(t, page, setupGroupLog, "first task in the task group '%s' should have the setup group log", taskName)
+		case lastTaskGroupTaskName:
+			const lastTaskGroupTaskLog = "smoke test is running the fourth task in the task group"
+			require.Contains(t, page, lastTaskGroupTaskLog, "last task in the task group '%s' should log expected output", taskName)
+			require.Contains(t, page, teardownGroupLog, "last task in the task group '%s' should have the teardown group log", taskName)
+		default:
+			// For all task group tasks that are neither the first or last,
+			// check that they do not run the setup group or teardown group.
+			require.NotContains(t, page, setupGroupLog, "task '%s' should not have the setup group log because it is not the first task in the task group", taskName)
+
+			require.NotContains(t, page, teardownGroupLog, "task '%s' should not have the teardown group log because it is not the last task in the task group", taskName)
+		}
+
+		if strings.Contains(taskName, "task_group") {
+			// For all tasks in the task group, check that they run the setup
+			// task and teardown task.
+			const setupTaskLog = "smoke test is running the setup task"
+			require.Contains(t, page, setupTaskLog, "task '%s' should log setup task log since it runs for every task in the task group", taskName)
+
+			const teardownTaskLog = "smoke test is running the teardown task"
+			require.Contains(t, page, teardownTaskLog, "teardown task should run for every task in the task group")
+		}
+
+	} else if mode == agent.PodMode {
+		const containerTaskLog = "container task"
+		assert.Contains(t, page, containerTaskLog, "should have found expected container task log")
+	}
+}
+
+// MakeSmokeRequest sends an authenticated smoke request to the smoke test app
+// server.
+func MakeSmokeRequest(ctx context.Context, params APIParams, method string, client *http.Client, url string) ([]byte, error) {
+	grip.Infof("Getting endpoint '%s'", url)
+
+	if !strings.HasPrefix(url, params.AppServerURL) {
+		url = strings.Join([]string{strings.TrimSuffix(params.AppServerURL, "/"), strings.TrimPrefix(url, "/")}, "/")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, method, url, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "making request for URL '%s'", url)
+	}
+	req.Header.Add(evergreen.APIUserHeader, params.Username)
+	req.Header.Add(evergreen.APIKeyHeader, params.APIKey)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting endpoint '%s'", url)
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading response body")
+	}
+	if resp.StatusCode >= 400 {
+		return body, errors.Errorf("got HTTP response code %d with error %s", resp.StatusCode, string(body))
+	}
+
+	return body, nil
+}
+
+func SmokeRunBinary(ctx context.Context, name, wd, bin string, cmdParts ...string) (*exec.Cmd, error) {
```

Comment:
Can we add a comment here and describe the inputs?

Reply:
Done.